### DISCUSSION TITLE
Color Wheel

### DISCUSSION THREAD

**Discussion by Mark198995** [arkypita/LaserGRBL#1689] [Ideas]
Are you able to put in the software an option for us to create a color wheel? Something similar to this [Video](https://www.youtube.com/watch?v=bzHqMhLgCQU) here. It would make deciding what settings are needed for materials alot easier. The recommended settings do not seem to work that well and I go through alot of product trying to figure out what settings will work best. I appreciate it in advance for taking the time to listen to my request. 
![color wheel](https://user-images.githubusercontent.com/96851410/147714645-2dfeb796-583f-4922-83b2-831ccb11f106.PNG)

**Comment by Tribo-2dot0** [comment]
I use this file to standardize power and speed:
![TestChart](https://user-images.githubusercontent.com/83982525/147720111-860ebc94-9d7e-4c29-a7ca-119ada7e0783.png)

**Reply by Mark198995** [reply]
is there a file i can export for my engraver to run this? thats what im really looking for.

**Comment by Tribo-2dot0** [comment]
What I do is open this image in LaserGRBL, burn it in the type of wood I want to use and choose the values I'm going to use.
Below are the samples I have in Curupixá and Marfim Arana (Brazilian woods).

![20211230_144628](https://user-images.githubusercontent.com/83982525/147776357-e87d5912-416d-4cdc-8f16-4f6655efbcf6.jpg)

![20211230_144636](https://user-images.githubusercontent.com/83982525/147776382-6d746cc4-520e-4e42-9ab3-e6d413b63124.jpg)

**Comment by Mark198995** [comment]
So on the second screen where you put the power do you do choose M3 constant or M4 dynamic. And for the power levels do you do the min as 5% and max 100%

**Comment by Tribo-2dot0** [comment]
For engraving I always use M4, S-MIN=0 and S-MAX=255

**Reply by Mark198995** [reply]
The 255 is that a percentage or the value? I ask cause I have a atomstack a5 20w (idk if relevant with second screen) and if I put a value of 255 that would be 26% of its power.

**Comment by Tribo-2dot0** [comment]
This will depend on the value set at $30 in the grbl settings. I think it comes by default 255 (others use 1000), so in this case 255=100%

![Captura de Tela (94)](https://user-images.githubusercontent.com/83982525/147779512-515fb1d2-2e6a-4d06-bee2-1254e18c3d5b.png)

**Reply by Mark198995** [reply]
Thanks a ton

**Comment by arkypita** [comment]
Correct use of grayscale test:
- first engrave it using the full S range of your laser (Smin = 0 and Smax = $30) at a medium speed (i.e 2000)
- if black it is too dark (or carved) try higher speed
- if black is too light, try lower speed

Now check your result: you may have more than one slice white, and more than one slice of the same black, but you can easily recognize a range of power where you can see difference between slices: this is your "good range".

Repeat the test using the previous speed but "correct" Smin and Smax to match the "good range".
I mean: assuming your laser use 0/1000 and your test show "good range" between 20% and 90% set SMin = 200 and SMax = 900.

**Reply by Tribo-2dot0** [reply]
I liked the formula... I've always used Smin=0, that's how the gray scale is more consistent...
A question about the value of $30... is there any difference when processing the G code using 1000 or 255? I've read that working with the value 1000 will have some more work at some point in the processing, does that match?
I ask because I always used 255, but for percentage calculation the value 1000 is much easier, since 1000=100%, 660=66%, etc.

**Reply by arkypita** [reply]
Controller use 8bit PWM modulation, so it is only able to give 255 different values of power. Bitmap images use 255 shades of gray too.
If your brain think like a computer 255 is much more natural and easier.

1000 is a number easy for percentage calculation.

Someone can use the power in milliwatt of his laser (i.e. 5500) so he can have different engraver with different laser modules, and use the same numbers to express the same power.

Your choice. https://lasergrbl.com/wp-content/uploads/2020/05/Grbl-Configuration-ENG.pdf

Fun fact: in origin GRBL firmware was developed for CNC so $30 was used to express spindle speed RPM at maximum power.

**Comment by gisep** [comment]
Hi guys,
i made this some time ago using velocity, density and greyscale.

![riempimento-velocità-potenza](https://user-images.githubusercontent.com/11257983/149616109-0b907fd4-50f1-45b6-b208-7c6b4f75e6fb.png)

To transform in g-code i made a lot of "appends", one for each Vel/Density combination. Maybe it could be automated with excel or something similar. If it could be useful i can append here the .svg but i don't know how to do it.
